public/js/components/App.js

```diff
@@ -60,8 +60,8 @@ const App = React.createClass({
   },
 
   componentDidMount() {
-    this.shortcuts = new KeyShortcuts({ window });
-    this.shortcuts.on("CmdOrCtrl+P", this.toggleSourcesSearch);
+    const shortcuts = setupShortcuts(window);
```

Comment:
I don't know if you have time to try it right now, but another option is to put it in [react context](https://facebook.github.io/react/docs/context.html). This would make it available to all components and I think it an appropriate usage of it. The benefit is that it would be available in the first render, because now it's `null` and I've seen a few issues with that. However I think you also solved that by making it a function `getShortcuts` instead of passing the instance down directly.

In fact, you could just pass down a function `{ getShortcuts: () => this.shortcuts }` too which would solve the `null` problem. Is that mainly what you're trying to solve here?